### ISSUE TITLE
Redundant variable capture for _

### ISSUE THREAD

**Issue by jrp2014** [ndmitchell/hlint#935]
The following `Warning: Redundant variable capture` is correct:
```
Found
_ <- sequence_ [primTrue, primFalse]
Perhaps
sequence_ [primTrue, primFalse]
```
but following it triggers a compiler warning, suggesting the you should revert to the original.

This is getting into matters of taste, but the original version suggests that you know what you are doing and the redundancy should not be warned about.  If the `_` was a variable `x`, the warning should stand.

**Comment by ndmitchell** [comment]
What `sequence_` are you using? What is the compiler warning?

**Comment by jrp2014** [comment]
I was confusing this with a different compiler warning.  A small example:
```
f :: [String]
f = do
  _ <- sequence_ ["xyz", "def"]
  x <- sequence_ ["xyz", "def"]
  sequence_ ["xyz", "def"]
  return "uvy"
```
results in
```
% ghci -Wall Frag.hs
GHCi, version 8.8.3: https://www.haskell.org/ghc/  :? for help
[1 of 1] Compiling Main             ( Frag.hs, interpreted )

Frag.hs:5:3: warning: [-Wunused-matches] Defined but not used: ‘x’
  |
5 |   x <- sequence_ ["xyz", "def"]
  |   ^


% hlint Frag.hs 
Frag.hs:4:3-31: Warning: Redundant variable capture
Found:
  _ <- sequence_ ["xyz", "def"]
Perhaps:
  sequence_ ["xyz", "def"]

1 hint
```

I think that it is still debatable that this should be warned as a "redundant variable capture".   At least it could be a suggestion.

**Comment by ndmitchell** [comment]
Doing `_ <- sequence_ ...` is super odd, because you've bound a variable of type `()` to `_`. That's a very unusual thing to do. You could legitimately say that everything that HLint says is a suggestion rather than a warning, but this seems a perfectly sensible thing to recommend changing to me.

**Comment by jrp2014** [comment]
Thinking about this further, I can see that.  Let's leave things be.  Thanks.
